.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,189 @@
+              -e "s@MAJOR 0@MAJOR ${{env.MAJOR}}@g" \
+              -e "s@MINOR 0@MINOR ${{env.MINOR}}@g" \
+              -e "s@BUILD 0@BUILD ${{ env.BUILD_ID }}@g" \
+              src/OSPSuite.SimModelSolver_CVODES/version.h
+
+      - name: Build Debug and Release
+        run: ./buildNix.sh ${{matrix.platform}}  ${{env.APP_VERSION}}
+
+      - name: Push nupkg as artifact
+        # if it is a push to a branch
+        if: github.event_name == 'push' && github.ref_name != github.event.repository.default_branch
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{matrix.nuget_package}}
+          path: ./*.nupkg
+
+      - name: Publish to GitHub registry
+        # if it is a merge to default branch
+        if: github.event_name == 'push' && github.ref_name == github.event.repository.default_branch
+        run: dotnet nuget push ./*.nupkg --source https://nuget.pkg.github.com/${{github.repository_owner}}/index.json --api-key ${{ secrets.GITHUB_TOKEN }}
+
+  build-windows-x64:
+    runs-on: windows-latest
+    needs: get-version
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: define env variables
+        run: |
+          echo "APP_VERSION=${{needs.get-version.outputs.APP_VERSION}}" | Out-File -FilePath $env:GITHUB_ENV -Append
+          echo "BUILD_ID=${{needs.get-version.outputs.BUILD_ID}}" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+      - name: Restore dependencies
+        run: |
+          nuget sources add -username Open-Systems-Pharmacology -password ${{ secrets.GITHUB_TOKEN }} -name OSP-GitHub-Packages -source "https://nuget.pkg.github.com/Open-Systems-Pharmacology/index.json"
+          dotnet restore --source https://ci.appveyor.com/nuget/ospsuite-utility --source https://ci.appveyor.com/nuget/ospsuite-bddhelper --source https://www.nuget.org/api/v2/
+
+      - name: Update version.h
+        shell: pwsh
+        run: |
+          (get-content src\OSPSuite.SimModelSolver_CVODES\version.h) | foreach-object {$_ -replace "RELEASE 0", "RELEASE ${{env.RELEASE}}" -replace "MAJOR 0", "MAJOR ${{env.MAJOR}}" -replace "MINOR 0", "MINOR ${{env.MINOR}}" -replace "BUILD 0", "BUILD ${{ env.BUILD_ID }}" } | set-content src\OSPSuite.SimModelSolver_CVODES\version.h
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v2
+
+      - name: build release
+        run: msbuild OSPSuite.SimModelSolver.CVODES.sln /p:Configuration=Release /p:Platform=x64
+
+      - name: build debug
+        run: msbuild OSPSuite.SimModelSolver.CVODES.sln /p:Configuration=Debug /p:Platform=x64
+
+      - name: Test
+        run: |
+          nuget install NUnit.ConsoleRunner -Version 3.19.2 -DirectDownload -OutputDirectory .
+          ./NUnit.ConsoleRunner.3.19.2/tools/nunit3-console.exe Build/Release/x64/OSPSuite.SimModelSolver_CVODES.Tests.Release.x64.dll
```

Comment:
Why is this different from the windows tests in other cpp packages?
E.g. compare
https://github.com/Open-Systems-Pharmacology/OSPSuite.FuncParser/blob/00d579344a682fda00b0a065c77075bb9a12e7f7/.github/workflows/build-and-publish.yml#L166-L167
```
      - name: Test
        run: dotnet test --no-build --no-restore --configuration:Release
```

Once all cpp repos are migrated to GH actions, I would try to make this workflow reusable and move it to the global Workflows repo. Differences between repos will make it harder.

Reply:
FuncParser tests are .NET but these are native C++

Reply:
True... now I remember that I did not migrate the tests from C++/CLI to PInvoke (just too much work for nothing)